### PR TITLE
ci: Fix version details

### PR DESCRIPTION
**Pull request by urutva**
### Description

If the tree is not dirty or a tag is available on the commit then
`mbed-greentea` test was failing. In this case, use the version as
returned by `setuptools-scm`.

### Pull request type

    [x] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

**Review by jainvikas8 — APPROVED** [review]

File: packages/mbed-greentea/test/mbed_gt_cli.py [hunk 1]
```diff
@@ -64,7 +64,7 @@ def test_get_greentea_version(self):
 
         self.assertIs(type(version), str)
         extra_info_pos = version.find('dev')
-        version = version[:extra_info_pos-1]
+        version = version[:extra_info_pos-1] if extra_info_pos > 0 else version
```
Comment (Patater) [inline packages/mbed-greentea/test/mbed_gt_cli.py:67]: Why can't we always use the `setuptools-scm` provided version? I don't see why we need `extra_info_pos`.

Comment (urutva) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: @Patater `setuptools-scm` returns version as `version.dev*` when the tree is dirty or the commit doesn't contain a tag. These tests are run for every PR and not every PR might have a tag associated with it. Therefore we need to filter out extra information appended to version by `setuptools-scm`.

Comment (Patater) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: Are there always three numbers present in the front of the version given by setuptools-scm?

Comment (urutva) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: yeah, the Default versioning scheme is explained here https://pypi.org/project/setuptools-scm/

Comment (Patater) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: So, we don't need to check for `dev` at all then, do we? The numbers are always present in the front and we can use those for a,b,c, no matter what comes later.

Comment (urutva) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: @Patater sure, but test fails if `split()` returns more than 3 digits. That's why we only extract version information if `setuptools_scm` adds `dev*` to the version.

Comment (Patater) [reply packages/mbed-greentea/test/mbed_gt_cli.py:67]: That's a problem with the way the test is written.

Try this, to ignore the later values:
```
- a,b,c = version.split('.')
+ a,b,c,_ = version.split('.')
```

**Review by jainvikas8 — APPROVED** [review]

**Review by Patater — APPROVED** [review]
Nice

**Pull request merged**